### PR TITLE
Kafka Utility functions

### PR DESCRIPTION
**Pull request by somil98**

### CODE REVIEW

File: kafka/kafkaConsumer.go [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+	RetryDuration                   time.Duration // default to 24 hours
+	offsetCommitMessageInterval     int           // default to 1000
+	lastOffsetCommitMessageInterval int
+	ReplayMode                      bool
+	ReplayFrom                      time.Duration //duration - defaults to 1h
+	ReplayType                      ReplayType
+	ReplyCompletionChannel          chan bool
+}
+
+func (kc *Consumer) applyCustomConfig(customConfig map[string]interface{}) {
```
Comment (somil98) [inline kafka/kafkaConsumer.go]: Why is this needed if we have SetConsumerCustomConfig

File: kafka/kafkaConsumer.go [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+			if e.Timestamp.After(consumerStartTime) {
+				return true
+			}
+		}
+		processor.ProcessMessage(&Message{Data: e.Value, TopicPartition: e.TopicPartition, Timestamp: e.Timestamp})
+		//kc.logger.LogDebug(fmt.Sprintf("Message on %s %s: %s Headers: %v", kc.InstanceID,
+		//	e.TopicPartition, string(e.Value), e.Headers))
+		kc.commitOffset()
+	case kafka.Error:
+		// Errors should generally be considered
```
Comment (xtroncode) [inline kafka/kafkaConsumer.go:264]: if this is true then why exit?

File: kafka/kafkaproducer.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+			for k, v := range customConfig {
+				kp.config.SetKey(k, v)
+			}
+		}
+	}
+}
+
+//ProducerLogger sets the logger for consul
+//Defaults to consul logger
+func ProducerLogger(customLogger *gologger.CustomLogger) ProducerOption {
```
Comment (xtroncode) [inline kafka/kafkaproducer.go]:
```suggestion
func SetProducerLogger(customLogger *gologger.CustomLogger) ProducerOption {
```

**Review by xtroncode — DISMISSED** [review]

File: kafka/kafkaproducer.go (hunk 3 above)
Comment (somil98) [reply kafka/kafkaproducer.go]: Done

**Review by xtroncode — DISMISSED** [review]

**Review by xtroncode — APPROVED** [review]
